### DISCUSSION TITLE
prisma openssl error

### DISCUSSION THREAD

**Discussion by codewithsisk** [prisma/prisma#19341] [Q&A] [answered]
iam trying to deploy a nestjs with prisma to fly.io the image seems to build successfully how ever iam getting this error causing the application failing to start. 
```
2023-05-18T15:29:05.199 app[5683700b74378e] jnb [info] prisma:warn Prisma failed to detect the libssl/openssl version to use, and may not work as expected. Defaulting to "openssl-1.1.x".

2023-05-18T15:29:05.199 app[5683700b74378e] jnb [info] Please manually install OpenSSL via `apt-get update -y && apt-get install -y openssl` and try installing Prisma again. If you're running Prisma on Docker, you may also try to replace your base image with `node:lts-slim`, which already ships with OpenSSL installed.
```
and yes the base image node:lts-slim

**Comment by joheee** [comment, marked as answer]
I solved the issue by installing openssl at the dockerfile

take a look at the command below is implemented for all the staging step

    RUN apk add --no-cache openssl

check my dockerfile below

    FROM node:20-alpine As development
    WORKDIR /usr/src/app
    RUN apk add --no-cache openssl
    COPY --chown=node:node package*.json ./
    RUN npm ci
    COPY --chown=node:node . .
    RUN npx prisma generate
    USER node
    
    FROM node:20-alpine As build
    WORKDIR /usr/src/app
    RUN apk add --no-cache openssl
    COPY --chown=node:node package*.json ./
    COPY --chown=node:node --from=development /usr/src/app/node_modules ./node_modules
    COPY --chown=node:node . .
    RUN npm run build
    ENV NODE_ENV production
    RUN npm ci --only=production && npm cache clean --force
    USER node
    
    FROM node:20-alpine As production
    RUN apk add --no-cache openssl
    COPY --chown=node:node --from=build /usr/src/app/node_modules ./node_modules
    COPY --chown=node:node --from=build /usr/src/app/dist ./dist
    COPY --chown=node:node --from=build /usr/src/app/public ./public
    CMD ["node", "dist/main.js"]

Follow me for more
- [github][1]
- ⁠[instagram][2]
- ⁠[medium][3]
- [youtube][4]


  [1]: https://github.com/joheee
  [2]: https://instagram.com/johedotcom
  [3]: https://medium.com/@johevinblesstowi07
  [4]: https://youtube.com/@johedotcomm

**Reply by RafalKielbasa** [reply]
It helped me a lot, thank you.

**Reply by Foehrenbach** [reply]
thanks!!!

**Reply by hungnp-zinza** [reply]
Thanks.

**Reply by BeThankxz** [reply]
Thanks !!!

**Reply by teammy** [reply]
Thank!!!

**Reply by rramosruiz** [reply]
Thank you!!!

**Reply by fearandesire** [reply]
Thank you

**Reply by DarkSmile92** [reply]
hero! Thanks!

**Reply by Shantodotdev** [reply]
Thank you bro!

**Reply by Med5KDB** [reply]
Thanks a lot! 🙌
You're a SuperHero! 🫵

**Reply by TSP-Dev** [reply]
last alpine has a problem, but with 3.20 it's ok, and no need to install openssl

**Reply by Abdev1205** [reply]
Thank you so much bro 🔥

**Reply by lucasczan** [reply]
thank you bro!

**Comment by SeventyB** [comment]
node:lts-slim looks like the latest docker image and OpenSSL may not the only change there which may produce other and more tricky issues within your code. Also the image will be changed and pushed in the future once again and you may get another issues. So it is better to stick to the specific docker image when building your app and switch to the next one intentionally and manually. To do that just update FROM statement in your docker file to something like this (this is an example and this image may not have openssl as well, check the tags [here](https://hub.docker.com/_/node/tags?name=slim)):

`FROM node:23.4.0-slim`

This way you will be ensured that the "base" for your code is always the same. If you want to get the latest patches automatically change it to `FROM node:23.4-slim` and for the patches and minor version updates `FROM node:23-slim`. This is not a docker feature but the suggested tagging schema for images and the same image can be under the different tags.

**Comment by hpohlmeyer** [comment]
If anyone is looking for a reason why this breaks now: https://github.com/nodejs/docker-node/issues/1827.
The alpine 3.16 images should still work:

```diff
- FROM node:20-alpine
+ FROM node:20-alpine3.16
```

**Reply by onahkenneth** [reply]
Thank you
